oauth2-server/src/main/java/com/clouway/oauth2/token/TokenRequest.java

```diff
@@ -1,44 +1,110 @@
 package com.clouway.oauth2.token;
 
+import com.clouway.oauth2.DateTime;
+import com.clouway.oauth2.Identity;
+import com.clouway.oauth2.client.Client;
+import com.google.common.base.Objects;
+
+import java.util.Map;
+import java.util.Set;
+
 /**
  * @author Ivan Stefanov <ivan.stefanov@clouway.com>
  */
 public class TokenRequest {
-  public final String grantType;
-  public final String code;
-  public final String refreshToken;
-  public final String clientId;
-  public final String clientSecret;
+  public final GrantType grantType;
+  public final Client client;
+  public final Identity identity;
+  public final Set<String> scopes;
+  public final DateTime when;
+  public final Map<String, String> params;
 
-  public TokenRequest(String grantType, String code, String refreshToken, String clientId, String clientSecret) {
+  public TokenRequest(GrantType grantType, Client client, Identity identity, Set<String> scopes, DateTime when, Map<String, String> params) {
```

Comment:
This constructor should be removed too and only builder usage to be enforced. 

Passing of many params as method arguments is similar as passing of many params to the constructor.

Reply:
Removed.